Zend/tests/partial_application/variation_strict_001.phpt

```diff
@@ -0,0 +1,21 @@
+--TEST--
+Partial application variation: strict_types declared
+--FILE--
+<?php
+declare(strict_types=1);
+
+function foo(int $int) {
+    var_dump($int);
+}
+
+$foo = foo(?);
+
+try {
+    $foo("42");
+} catch (TypeError $ex) {
+    printf("%s\n", $ex->getMessage());
+}
```

Comment:
I think for this test what we really want is two files, one strict where the partial is created and another weak where it is called.

What would the expected strictness be in that case (and is it the same for arguments bound in strict file and those applied in weak file)?